modules/howtos/examples/collection-manager.js

```diff
@@ -3,66 +3,92 @@
-        console.log("The scope already exists");
-      }
-    }
-    // end::create-scope[]
+  await userMgr.upsertUser({
+    username: "scope_admin",
+    password: "password",
+    displayName: "JS Manage Scopes [travel-sample:*]",
+    roles: [
+      { name: "scope_admin", bucket: "travel-sample" },
+      { name: "data_reader", bucket: "travel-sample" }
+    ]
+  })
+  // end::scopeAdmin[]
 
-    // tag::create-collection[]
-    try {
-      await collectionMgr.createCollection("example-collection", "example-scope");
-    }
-    catch (e) {
-      if (e instanceof couchbase.CollectionExistsError) {
-        console.log("The collection already exists");
-      }
-      // else if (e instanceof couchbase.ScopeNotFoundError) {
-      //   console.log("The scope does not exist");
-      // }
-    }
-    // end::create-collection[]
+  const cluster = new couchbase.Cluster("couchbase://localhost", {
+    username: "scope_admin",
+    password: "password",
+  });
 
-    // tag::drop-collection[]
-    try {
-      await collectionMgr.dropCollection("example-collection", "example-scope");
+  // tag::create-collection-manager[]
+  const bucket = cluster.bucket("travel-sample");
+  const collectionMgr = bucket.collections();
+  // end::create-collection-manager[]
+
+  // tag::create-scope[]
+  try {
+    await collectionMgr.createScope("example-scope");
+  }
+  catch (e) {
+    if (e instanceof couchbase.ScopeExistsError) {
+      console.log("The scope already exists");
     }
-    catch (e) {
-      // if (e instanceof couchbase.CollectionNotFoundError) {
-        console.log("The collection does not exist");
-      // }
-      // else if (e instanceof couchbase.ScopeNotFoundError) {
-      //   console.log("The scope does not exist");
-      // }
+    else {
+      throw e
     }
-    // end::drop-collection[]
+  }
+  // end::create-scope[]
 
-    // tag::drop-scope[]
-    try {
-      await collectionMgr.dropScope("example-scope");
+  // tag::create-collection[]
+  try {
+    await collectionMgr.createCollection("example-collection", "example-scope");
+  }
+  catch (e) {
+    if (e instanceof couchbase.CollectionExistsError) {
+      console.log("The collection already exists");
     }
-    catch (e) {
-      // if (e instanceof couchbase.ScopeNotFoundError) {
-        console.log("The scope does not exist");
-      // }
+    else {
+      throw e
     }
-    // end::drop-scope[]
+    // else if (e instanceof couchbase.ScopeNotFoundError) {
+    //   console.log("The scope does not exist");
+    // }
+  }
+  // end::create-collection[]
+
+  // tag::drop-collection[]
+  try {
+    await collectionMgr.dropCollection("example-collection", "example-scope");
+  }
+  catch (e) {
+    // if (e instanceof couchbase.CollectionNotFoundError) {
+      console.log("The collection does not exist");
+    // }
+    // else if (e instanceof couchbase.ScopeNotFoundError) {
+    //   console.log("The scope does not exist");
+    // }
```

Comment:
And these lines too

Reply:
Yeah, these were commented previously:

```
❯ git blame modules/howtos/examples/collection-manager.js | grep '// else if '
0aeba097 (Hakim Cassimally 2021-03-19 17:04:06 +0000 36)       // else if (e instanceof couchbase.ScopeNotFoundError) {
0aeba097 (Hakim Cassimally 2021-03-19 17:04:06 +0000 50)       // else if (e instanceof couchbase.ScopeNotFoundError) {
❯ git show --no-patch 0aeba097
commit 0aeba09726bd706b74eeb3caab417a995fd24d27 (origin/doc-7968-collection-management)
Author: Hakim Cassimally <hakim.cassimally@couchbase.com>
Date:   Fri Mar 19 17:04:06 2021 +0000

    DOC-7968 document Collection Management

    NB: some of the error checking is commented out, pending JSCBC-858
```

I have a watch on https://issues.couchbase.com/browse/JSCBC-858
so hoping to update by 3.2.

Reply:
ah gotcha, maybe add a comment like: `TODO: uncomment code once JSCBC-858 is resolved` or something along those lines.

Reply:
ok, new commit using Asciidoc's tag filtering, how's that?

Reply:
Love this approach! 👍